Elaborate on the control flow behavior of try-catch-finally (#17954)
mdn/content: files/en-us/web/javascript/reference/statements/try...catch/index.md

  - : The statements to be executed.
- `catch_statements`
  - : Statement that is executed if an exception is thrown in the `try`-block.
- `exception_var` {{optional_inline}}

>   - : An optional identifier to hold the caught exception for the associated `catch`-block. If the `catch` block does not utilize the exception's value, you can omit the `exception_var` and its surrounding parentheses, as `catch {...}`.

Reviewer: ```suggestion
  - : An optional identifier to hold the caught exception for the associated `catch` block. If the `catch` block does not utilize the exception's value, you can omit the `exception_var` and its surrounding parentheses, as `catch {...}`.
```

- `finally_statements`
  - : Statements that are executed before control flow exits the `try...catch...finally` construct. These statements execute regardless of whether an exception was thrown or caught.

## Description

> The `try` statement always starts with a `try`-block. Then, a `catch`-block, a `finally`-block, or both must be present. This gives us three forms for the `try` statement:

Reviewer: ```suggestion
The `try` statement always starts with a `try` block. Then, a `catch` block, a `finally` block, or both must be present. This gives us three forms for the `try` statement:
```

- `try...catch`
- `try...finally`
- `try...catch...finally`

> Unlike other constructs like [`if`](/en-US/docs/Web/JavaScript/Reference/Statements/if...else) or [`for`](/en-US/docs/Web/JavaScript/Reference/Statements/for), the `try`, `catch`, and `finally` blocks must be _blocks_, instead of single statements.

Reviewer: ```suggestion
Unlike other constructs such as [`if`](/en-US/docs/Web/JavaScript/Reference/Statements/if...else) or [`for`](/en-US/docs/Web/JavaScript/Reference/Statements/for), the `try`, `catch`, and `finally` blocks must be _blocks_, instead of single statements.
```

```js example-bad
try doSomething(); // SyntaxError
catch (e) console.log(e);
```

A `catch`-block contains statements that specify what to do if an exception
is thrown in the `try`-block. If any statement within the
`try`-block (or in a function called from within the `try`-block)
throws an exception, control is immediately shifted to the `catch`-block. If
no exception is thrown in the `try`-block, the `catch`-block is
skipped.

> The `finally`-block will always execute before control flow exits the `try...catch...finally` construct. It always executes, regardless of whether an exception was thrown or caught.

Reviewer: ```suggestion
The `finally` block will always execute before control flow exits the `try...catch...finally` construct. It always executes, regardless of whether an exception was thrown or caught.
```

}
```

> Control flow statements (`return`, `throw`, `break`, `continue`) in the `finally` block will "mask" any completion value of the `try`-block or `catch`-block. In this example, the `try`-block tries to return 1, but before returning, the control flow is yielded to the `finally`-block first, so the `finally`-block's return value is returned instead.

Reviewer: ```suggestion
Control flow statements (`return`, `throw`, `break`, `continue`) in the `finally` block will "mask" any completion value of the `try` block or `catch` block. In this example, the `try` block tries to return 1, but before returning, the control flow is yielded to the `finally` block first, so the `finally` block's return value is returned instead.
```
